apps/src/webapp/webapp.js

```diff
@@ -1618,6 +1685,28 @@ var checkFinished = function () {
   return false;
 };
 
+// TODO(dave): move this logic to dashboard.
```

Comment:
agree, I think dashboard should pass the URL to the webapp code in the config block. and the webapp code should fail gracefully if it is not passed in (or if it is being run without the dashboard).